pkg/controller/acmeorders/sync.go

```diff
@@ -265,13 +267,14 @@ func (c *controller) createOrder(ctx context.Context, cl acmecl.Interface, o *cm
 	log.V(logf.DebugLevel).Info("order URL not set, submitting Order to ACME server")
 
 	dnsIdentifierSet := sets.New[string](o.Spec.DNSNames...)
-	if o.Spec.CommonName != "" {
+	ipIdentifierSet := sets.New[string](o.Spec.IPAddresses...)
+	if len(validation.IsValidIP(field.NewPath(""), o.Spec.CommonName)) == 0 {
+		ipIdentifierSet.Insert(o.Spec.CommonName)
+	} else if len(validation.IsFullyQualifiedDomainName(field.NewPath(""), o.Spec.CommonName)) == 0 {
 		dnsIdentifierSet.Insert(o.Spec.CommonName)
 	}
```

Comment:
```suggestion
	switch {
	case o.Spec.CommonName == "":
	case net.ParseIP(o.Spec.CommonName) != nil:
		ipIdentifierSet.Insert(o.Spec.CommonName)
	case len(validation.IsFullyQualifiedDomainName(nil, o.Spec.CommonName)) == 0:
		dnsIdentifierSet.Insert(o.Spec.CommonName)
	}
```

WDYT of using a switch instead? I find it easier to read, and we should also keep the early exit if CN is not set.

And I think we can use the Go stdlib function to check for IP instead of the apimachinery wrapper function.